chapter15/qli/ex1.pl

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use Scalar::Util qw(looks_like_number);
+use 5.010001;
+
+my $rand = int(rand(100))+1;
+
+print "Please guess a number\n";
+while(1){
+    chomp (my $input = <STDIN>) ;
+   
+    if( $input =~ /exit/i ){
+      print "Good bye!\n";
+      exit 1;
+    }elsif( !looks_like_number( $input ) ){
+      print "Not number and unsupported cmd.\n";
+      next;  
+    }
+   
+    given ( $input  ) {
+        when ( $_ == $rand ){
```

Comment:
This could be put in the "default" block since it's what will happen should it fail the other two and then you don't need this condition.